xla/pjrt/gpu/gpu_helpers.h

```diff
@@ -57,6 +57,10 @@ struct GpuAllocatorConfig {
   // fragmentation, allowing more of the total memory to be used. If false, the
   // allocator will allocate more memory as allocations are requested.
   bool preallocate = true;
+
+  // Amount of collective memory (ncclMemAlloc) to reserve. If this value is 0,
+  // collective memory will not be used.
+  size_t collective_memory_size = 1 * (1LL << 30);
```

Comment:
Correct me if I'm wrong should this size be the same as NCCL_BUFFSIZE?
If user will set different value here wouldn't it led to runtime failure?
If so should this parameter be boolean and could we get actual size from env variable?

Reply:
This size we need here is determined by the peak memory usage of NCCL collective buffers in the model which is calculated during buffer assignment.

For example, an 8-gpu GPT5B needs around 400mb from looking at the dumped after-optimizations-buffer-assignment.txt file:
```
allocation 318: size 428487776, color 1, preallocated-temp:
 value: <1000 reduce-scatter.15 @1> (size=16777216,offset=896): bf16[16384,512]{0,1:S(1)}
 value: <1004 reduce-scatter.16 @1> (size=12582912,offset=33555328): bf16[3,32,128,512]{2,1,0,3:S(1)}
 . . .
 value: <4489 all-gather-start.10{1} @1> (size=411705344,offset=896): bf16[50257,4096]{0,1:S(1)}
```

Also, `ncclMemAlloc` aligns the size to the nearest 512mb so this number should be a multiple of 512mb to avoid wasting memory.